src/sv-pipeline/04_variant_resolution/scripts/integrate_resolved_vcfs.py

```diff
@@ -0,0 +1,107 @@
+#!/bin/env python
+
+import argparse
+import sys
+from typing import Optional, List, Text, Dict, Set, Callable
+
+import pysam
+
+MEMBERS_KEY = "MEMBERS"
+UNRESOLVED_KEY = "UNRESOLVED"
+
+
+def update_header(header: pysam.VariantHeader) -> None:
+    header.add_line('##INFO=<ID=UNRESOLVED,Number=0,Type=Flag,Description="Variant is unresolved.">')
+    header.add_line('##INFO=<ID=UNRESOLVED_TYPE,Number=1,Type=String,Description=\"Class of unresolved variant.\">')
+
+
+def is_unresolved(record: pysam.VariantRecord):
+    return record.info.get(UNRESOLVED_KEY, None)
+
+
+def is_resolved(record: pysam.VariantRecord):
+    return not is_unresolved(record)
+
+
+def get_members(record: pysam.VariantRecord):
+    return list(record.info[MEMBERS_KEY]) if isinstance(record.info[MEMBERS_KEY], tuple) \
+        else [record.info[MEMBERS_KEY]] if record.info[MEMBERS_KEY] is not None \
+        else list()
+
+
+def get_vids_and_members_sets(vcf: pysam.VariantFile,
+                              predicate: Callable) -> Dict:
+    unresolved_vids_set = set()
+    unresolved_members_set = set()
+    for r in vcf:
+        if predicate(r):
+            unresolved_vids_set.add(r.id)
+            unresolved_members_set.update(get_members(r))
+    vcf.reset()
+    return unresolved_vids_set, unresolved_members_set
+
+
+def write_vcf(header: pysam.VariantHeader,
+              all_vcf: pysam.VariantFile,
+              inv_vcf: pysam.VariantFile,
+              inv_resolved_vids_set: Set,
+              inv_resolved_members_set: Set,
+              all_unresolved_vids_set: Set,
+              all_unresolved_members_set: Set) -> None:
+    sys.stdout.write(str(header))
+    for r in all_vcf:
+        if r.id not in all_unresolved_vids_set or r.id not in inv_resolved_members_set:
+            # Resolved in ALL vcf, or unresolved in both VCFs
+            sys.stdout.write(str(r))
+        else:
+            sys.stderr.write(f"all excluded: {r.id}\n")
+    for r in inv_vcf:
+        if r.id in inv_resolved_vids_set:
+            # Resolved variant in the INV vcf
+            members = get_members(r)
+            sys.stderr.write(f"inv considered: {r.id}\n")
```

Comment:
Do these logging lines create a large stderr, or are INV records rare enough that it's not an issue?

Reply:
Accidentally left those in